packages/story-editor/src/app/media/utils/useFFmpeg.js

```diff
@@ -282,6 +282,81 @@ function useFFmpeg() {
     [getFFmpegInstance]
   );
 
+  /**
+   * Segment a video using FFmpeg.
+   *
+   * @param {File} file Original video file object.
+   * @param {string} segmentTime number of secs to split the video into.
+   * @return {Promise<File[]>} Segmented video files .
+   */
+  const segmentVideo = useCallback(
+    async (file, segmentTime, fileLength) => {
+      //eslint-disable-next-line @wordpress/no-unused-vars-before-return -- False positive because of the finally().
+      const trackTiming = getTimeTracker('segment_video');
+      let ffmpeg;
+      try {
+        ffmpeg = await getFFmpegInstance(file);
+        const type = file?.type || MEDIA_TRANSCODED_MIME_TYPE;
+        const ext = getExtensionFromMimeType(type);
+        const outputFileName = getFileBasename(file) + '_%03d.' + ext;
+        const keyframes = [];
+        for (let i = segmentTime; i < fileLength; i += segmentTime) {
+          keyframes.push(i);
+        }
+        const segmentTimes = keyframes.join(',');
+
+        await ffmpeg.run(
+          '-i',
+          file.name,
+          '-c',
+          'copy',
+          '-map',
+          '0',
+          '-force_key_frames',
+          `${segmentTimes}`,
+          '-f',
+          'segment',
+          '-segment_times',
+          `${segmentTimes}`,
+          '-segment_time_delta', //account for possible roundings operated when setting key frame times.
+          `${(1 / (2 * FFMPEG_CONFIG.FPS[1])).toFixed(2)}`, //
```

Comment:
Empty comment?

Reply:
will remove --- was going to comment the resulting value as might be easier to read that line